makefiles/dependencies_debug.inc.mk

```diff
@@ -0,0 +1,51 @@
+# Debug targets to evaluate dependency migrations
+#
+# The goal is to get the value of variables used for dependency resolution
+# and the impact of refactoring
+
+# Files output can be generated through `info-boards-supported` evaluation with
```

Comment:
This returns incorrect resolution in some cases if I'm understanding right from your PR description, right?

Reply:
The script returns the resolutions from the build system, but the build system returns incorrect parsing when doing `info-boards-supported` as it ignores the content of `boards/cpu/Makefile.include` that are currently required.

Reply:
Maybe we need to add a note saying they are different and that there is a "correct" one, similar to your PR comment.

Reply:
Sure I add this.

Reply:
I will flip the definitions to have the "normal" one first.